File: frontend/src/api/ocr.ts
import client from './client'

export interface OcrExtractResult {
  raw_text: string
  parsed: {
    trade_date?: string
    trade_time?: string
    direction?: string
    price?: number
    quantity?: number
  }
  error?: string
}

export const extractTradeFromImage = (file: File) => {
  const form = new FormData()
  form.append('file', file)
  return client.post<OcrExtractResult>('/ocr/extract', form)
}

File: backend/app/routers/ocr.py
from fastapi import APIRouter, UploadFile, File
from app.services.ocr_service import extract_text, parse_trade_from_text

router = APIRouter(prefix="/api/ocr", tags=["ocr"])


@router.post("/extract")
async def extract_trade_from_image(file: UploadFile = File(...)):
    """上传券商成交截图，OCR 识别并解析交易字段。"""
    content = await file.read()
    try:
        raw_text = extract_text(content)
    except RuntimeError as e:
        return {"raw_text": "", "parsed": {}, "error": str(e)}
    parsed = parse_trade_from_text(raw_text)
    return {"raw_text": raw_text, "parsed": parsed}
